cmd/lassie/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	lcli "github.com/filecoin-project/lassie/cli"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/urfave/cli/v2"
+)
+
+var log = logging.Logger("lassie")
+
+func main() {
+	// set up a context that is canceled when a command is interrupted
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// set up a signal handler to cancel the context
+	go func() {
```

Comment:
urfave should already handle all this I don't think you need this code

Reply:
Removing this did not end up letting the daemon finish running

Reply:
yeah, looks like urfave/cli/v2 backed out of signal handling:

> We've decided that the argument parsing library is not in the position to do signal handling, and instead allow users to pass a context to the cli which they can now cancel

so I think this is right

Reply:
Good find!